File: antabayanaka/issue_token.py
import os
import json
import pytz
from dotenv import load_dotenv
from api.cline import Cline
from api import keys
from ineryjs import Api, JsonRpc, RpcError, JsSignatureProvider

load_dotenv()

TOKEN_CONTRACT = os.environ.get("TOKEN_CONTRACT")
ISSUER_ACCOUNT = os.environ.get("ISSUER_ACCOUNT")
TOKEN_SYMBOL = os.environ.get("TOKEN_SYMBOL")
ISSUE_QUANTITY = os.environ.get("ISSUE_QUANTITY")
ISSUE_MEMO = os.environ.get("ISSUE_MEMO")

json_rpc = JsonRpc(os.environ.get("INERY_NODE_URL"))
signature_provider = JsSignatureProvider([os.environ.get("PRIVATE_KEY")])

api = Api(rpc=json_rpc, signature_provider=signature_provider)

async def issue_token(token_contract:str, issuer:str, quantity:str, memo:str):
    try:
        tx = await api.transact({
            "actions": [
                {
                    "account": token_contract,
                    "name": "issue",
                    "authorization": [
                        {
                            "actor": issuer,
                            "permission": "active"
                        }
                    ],
                    "data": {
                        "to": issuer,
                        "quantity": quantity,
                        "memo": memo
                    }
                }
            ]
        }, broadcast=True, sign=True)

        print("transaction details:")
        print(tx.processed)
        print("RPC transaction details:")
        print(tx.processed.action_traces[0].act)
    except RpcError as e:
        print("ERROR: Can't issue token symbol", TOKEN_SYMBOL)
        print("DETAILS:", e.details[0].message)

issue_token(TOKEN_CONTRACT, ISSUER_ACCOUNT, ISSUE_QUANTITY, ISSUE_MEMO)

File: wardy/issue_token.mjs
import { Api, JsonRpc, RpcError, JsSignatureProvider } from '../dist/index.js'
import * as dotenv from 'dotenv' 

dotenv.config()

const url = process.env.NODE_URL; // use your http_address with port : 88888
const json_rpc = new JsonRpc(url); // create new JsonRPC using your node url

const private_key = process.env.INERY_PKEY; // private key of your inery account
const account = process.env.INERY_ACC; //your account inery
const actor = process.env.INERY_ACC; // The Signer
const token = process.env.TOKEN;
const signature  = new JsSignatureProvider([private_key]); // create signature from private_key
const api = new Api({
    rpc: json_rpc,
    signatureProvider: signature
});

// issue Function on Smart Contract
async function issuetoken(to, quantity, memo){
    try{
        const tx = await api.transact({
            actions:[
                {
                  account,
                  name:"issue",
                  authorization:[
                        {
                            actor,
                            permission:"active"
                        }
                    ],
                    data:{
                        to, quantity, memo
                    }
                }
            ]
        },{broadcast:true,sign:true})

        console.log(tx) // output the tx to terminal, it's Json Object
        console.log(tx.processed.action_traces[0].console)
    }catch(error){
        console.log(error)
    }
}

issuetoken(actor, token, "issue form rpc")